File: velolabs_repos/lattis_android/app/src/main/java/com/lattis/ellipse/data/platform/mapper/AccountMapper.java
package com.lattis.ellipse.data.platform.mapper;

import androidx.annotation.NonNull;

import com.lattis.ellipse.data.network.base.AbstractDataMapper;
import com.lattis.ellipse.domain.model.Account;
import com.lattis.ellipse.presentation.dagger.qualifier.AccountType;

import javax.inject.Inject;

public class AccountMapper extends AbstractDataMapper<android.accounts.Account, Account>{

    private String accountType;

    @Inject
    public AccountMapper(@AccountType String accountType) {
        this.accountType = accountType;
    }

    @NonNull
    @Override
    public Account mapIn(@NonNull android.accounts.Account androidAccount) {
        return new Account(androidAccount.name, androidAccount.type);
    }

    @NonNull
    @Override
    public android.accounts.Account mapOut(@NonNull Account account) {
        return new android.accounts.Account(account.getAccountName(),accountType);
    }
}

File: velolabs_repos/operator-android/data/src/main/java/io/lattis/data/mapper/AccountMapper.kt
package io.lattis.data.mapper

import io.lattis.domain.models.Account
import javax.inject.Inject
import javax.inject.Named

class AccountMapper @Inject
constructor(@param:Named("AccountType") private val accountType: String) : AbstractDataMapper<android.accounts.Account, Account>() {

    override fun mapIn(androidAccount: android.accounts.Account?): Account {
        return Account(androidAccount?.name, androidAccount?.type)
    }

    override fun mapOut(account: Account?): android.accounts.Account? {
        return android.accounts.Account(account?.accountName, accountType)
    }
}
